manifests/mount.pp

```diff
@@ -29,6 +31,24 @@
 
   $repo = $name
 
+  # UID and GID map are stored in separate files and included in the config.
```

Comment:
Argh I see you do support a map for a single mount, sorry kind of missed that.

I see a lot of duplicated code here.

How about creating a new defined type.

```
cvmfs::id_map{name:
  map => $cvmfs_uid_map,
  location => '/etc/cvmfs/config.d/<whatver>
}
```
this can be called from the config.pp file for general cases and the individual cvmfs::mount types

It would also allow us to get rid of the 

```puppet
[ 'uid', 'gid' ].each |$_idt| {
    $_cvmfs_id_map = getvar("cvmfs_${_idt}_map")
```
which makes everything a bit unreadable. The type could just called twice for group and user.

With that given the template is quite so trivial the content could be replaced with a map and join but which ever is fine for the last bit.

Reply:
Interesting idea! 
I have implemented that now (completely untested, will test it also on a real node here and have to adapt the spec tests still). Does my last commit match what you had in mind?

Reply:
I've tested it on a fresh node here by now and also added a test for the new defined type :smile:.